independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/DependentContext.java

```diff
@@ -0,0 +1,60 @@
+package io.quarkus.arc.impl;
+
+import java.lang.annotation.Annotation;
+
+import jakarta.enterprise.context.Dependent;
+import jakarta.enterprise.context.spi.Contextual;
+import jakarta.enterprise.context.spi.CreationalContext;
+
+import io.quarkus.arc.InjectableBean;
+import io.quarkus.arc.InjectableContext;
+
+/**
+ * The built-in context for {@link jakarta.enterprise.context.Dependent}.
+ */
+class DependentContext implements InjectableContext {
+    @Override
+    public Class<? extends Annotation> getScope() {
+        return Dependent.class;
+    }
+
+    @Override
+    public <T> T get(Contextual<T> contextual, CreationalContext<T> creationalContext) {
+        if (creationalContext == null) {
+            return null;
+        }
+
+        T instance = contextual.create(creationalContext);
+        if (creationalContext instanceof CreationalContextImpl) {
+            // we can remove this `if` and cast unconditionally after https://github.com/jakartaee/cdi-tck/pull/452
```

Comment:
I wouldn't mind being paranoid and keeping the check but we can surely just blow up with an exception once that TCK issue goes through.

Reply:
We generally assume that all CCs are `CreationalContextImpl`, which I think is fair. I thought I can add the check here, because 1. it lets us pass one more TCK test :-), 2. the context object for the dependent scope is never used anywhere. It has to exist solely to satisfy the specification's demand that a context object exist.